pkg/cmd/cli/cmd/newapp.go

```diff
@@ -531,8 +533,20 @@ func handleRunError(c *cobra.Command, err error, fullName string) error {
 	}
 	switch t := err.(type) {
 	case newcmd.ErrRequiresExplicitAccess:
+		if t.Input.Token != nil && t.Input.Token.ServiceAccount {
+			return fmt.Errorf(`installing %q requires an 'installer' service account with project editor access
+
+WARNING: This will allow the pod to create and manage resources within your namespace -
+ensure you trust the image with those permissions before you continue.
+
+You can see more information about the image by adding the --dry-run flag.
+If you trust the provided image, include the flag --grant-install-rights.`, t.Match.Value)
+		}
```

Comment:
Let's queue that up for after - I agree though.  Is that ok?

On Fri, Nov 6, 2015 at 2:02 PM, Cesar Wong <notifications@github.com> wrote:

> In pkg/cmd/cli/cmd/newapp.go
> <https://github.com/openshift/origin/pull/5724#discussion_r44174382>:
>
> > @@ -531,8 +533,20 @@ func handleRunError(c *cobra.Command, err error, fullName string) error {
> >  	}
> >  	switch t := err.(type) {
> >  	case newcmd.ErrRequiresExplicitAccess:
> > +		if t.Input.Token != nil && t.Input.Token.ServiceAccount {
> > +			return fmt.Errorf(`installing %q requires an 'installer' service account with project editor access
> > +
> > +WARNING: This will allow the pod to create and manage resources within your namespace -
> > +ensure you trust the image with those permissions before you continue.
> > +
> > +You can see more information about the image by adding the --dry-run flag.
> > +If you trust the provided image, include the flag --grant-install-rights.`, t.Match.Value)
> > +		}
>
> @smarterclayton <https://github.com/smarterclayton> just one more nit
> about this message. It'd be nice if it told me that if I specify
> --grant-install-rights, a service account will be created for me if one
> doesn't already exist.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/openshift/origin/pull/5724/files#r44174382>.
>